### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by karolinachodera**

### CODE REVIEW

File: script.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    parentItem.remove();
+    items.splice(index, 1);
+    localStorage.setItem("items", JSON.stringify(items));
+
+    if(items.length === 0 && list) {
+        removeList();
+    }
+}
+
+function startItem(this: HTMLFormElement, e: Event): void {
```
Comment (biern-j) [inline script.ts:156]: Great! I also found an article with an explanation of how to add types to 'this'  [here](https://betterprogramming.pub/introduction-to-typescript-functions-this-object-and-overloads-6d03a90f1821).

File: script.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+class ItemObject {
+    constructor (name: string, category: string) {
+        this.name = name;
+        this.category = category;
+        this.id = index;
+        this.isChecked = false;
+    }
+}
+
+function moveToEnd(this: HTMLElement, e: Event): void{
```
Comment (biern-j) [inline script.ts]:
```suggestion
function moveToEnd(this: HTMLElement, e: Event): void {
```

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    // "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
+    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type. */
+    // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
+    // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
+
+    /* Completeness */
+    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
+    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
+  }
+}
```
Comment (biern-j) [inline tsconfig.json:16]: Do you need these commanded config rules? I get would rid of them.

File: node_modules/typescript/package.json
Comment (biern-j) [inline node_modules/typescript/package.json:1]: Please add the 'node_modules' folder to .gitignore.
I recommend reading [this article](https://www.freecodecamp.org/news/gitignore-file-how-to-ignore-files-and-folders-in-git/#contents) to get to know what kind of files and folders we should put into the .gitignore file.

Comment (biern-j) [reply node_modules/typescript/package.json:1]: When you will make this correction you will have to remove node_modules from git.
With this command you will delete node_modules from git and clear cached git history:
`$ git rm --cached .node_modules`

File: script.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+class ItemObject {
+    constructor (name: string, category: string) {
+        this.name = name;
+        this.category = category;
+        this.id = index;
+        this.isChecked = false;
+    }
+}
```
Comment (biern-j) [inline script.ts]: I would pass the types directly into the class. 

```suggestion
class ItemObject {
    name: string;
    category: string;
    id: number;
    isChecked: boolean;
    constructor (name: string, category: string) {
        this.name = name;
        this.category = category;
        this.id = index;
        this.isChecked = false;
    }
}
```

**Review by biern-j — CHANGES_REQUESTED** [review]
Nice work :) I hope this exercise helps you lear TypeScript.
We can talk about my comments on the next meeting.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (biern-j) [inline .gitignore:1]: This red icon means that you have an empty line at the end of the code. You can delete it ;)

File: script.ts (hunk 4 above)
Comment (karolinachodera) [reply script.ts]: I'm not sure if I combine these parts of the code, will I still have type/interface "ItemObject" defined? Or maybe defining an interface is not needed when using a class?

**Review by biern-j — APPROVED** [review]

**Pull request merged**